examples/dnsblock/dnsblock.lua

```diff
@@ -0,0 +1,89 @@
+--
+-- SPDX-FileCopyrightText: (c) 2024 Mohammad Shehar Yaar Tausif <sheharyaar48@gmail.com>
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+
+-- Filter DNS packets based on a blocklist
+
+local xt = require("xtable")
+local linux = require("linux")
+local string = require("string")
+local action = xt.action
+local family = xt.family
+
+local udp = 0x11
+local dns = 0x35
+
+local blacklist = {
+	"github.com",
+	"gitlab.com",
+}
+
+local function nop() end
+
+local function check_blacklist (name)
+	for _,v in ipairs(blacklist) do
+		if string.find(name, v) ~= nil then
+			return true
+		end
+	end
+	return false
+end
+
+local function get_domainname (skb, off)
+	local name = ""
+	local i = 0
+	while true do
+		local d = skb:getbyte(off + i)
+		if d == 0 then
+			break
+		end
+
+		i = i + 1
+		local len = d
+
+		while len > 0 do
+			name = name .. string.char(skb:getbyte(off + i))
+			i = i + 1
+			len = len - 1
+		end
+
+		name = name .. "."
+	end
+	return name
+end
+
+local function match_packet(skb, par)
+	print("match_packet called")
+
+	-- ip header
+	local thoff = par["thoff"]
+	local proto = skb:getuint8(9)
+
+	if proto == udp then
+		local dstport = linux.ntoh16(skb:getuint16(thoff + 2))
+		if dstport == dns then
+			local qoff = thoff + 20
+			local name = get_domainname(skb, qoff)
+			if check_blacklist(name) then
+				print("DNS query for " .. name .. " blocked\n")
+				return action.ACCEPT
+			end
+		end
+	end
+
+	return action.DROP
+end
+
+local match_ops = {
```

Comment:
this is quite a minor comment and totally up to you.. but it would look better to me to have use the netfilter module name on this table.. (e.g., `local dnsblock_mt = { ... }`) at the beginning of this module and then `function dnsblock_mt.match(...)` or suppressing the local variable here and doing:

```
xt.match{
	name = "dnsblock",
	...
}
```

both ways look better to me.. but again.. up to you ;-)